src/server.rs

```diff
@@ -421,6 +426,68 @@ async fn stream_classification_with_gen(
     Sse::new(event_stream).keep_alive(KeepAlive::default())
 }
 
+#[instrument(skip_all)]
+async fn stream_content_detection(
+    State(state): State<Arc<ServerState>>,
+    headers: HeaderMap,
+    request: Request,
+) -> Response {
+    let trace_id = Span::current().context().span().span_context().trace_id();
+    let headers = filter_headers(&state.orchestrator.config().passthrough_headers, headers);
+    info!(?trace_id, "handling content detection streaming request");
+
+    // Create input stream
+    let input_stream = request
+        .into_body()
+        .into_data_stream()
+        .map(|result| {
+            serde_json::from_slice::<StreamingContentDetectionRequest>(&result.unwrap())
+                .map_err(|e| orchestrator::Error::JsonError(e.to_string()))
+        })
```

Comment:
Let's call `validate()` here after deserialization. To use `?` and avoid `map_err`, add the `From` impl below to `orchestrator/errors.rs`

```suggestion
        .map(|result| {
            // Deserialize and validate messages
            let msg = serde_json::from_slice::<StreamingContentDetectionRequest>(&result.unwrap())?;
            msg.validate()?;
            Ok(msg)
        })
```

Add to `orchestrator/errors.rs`:
```rust
impl From<ValidationError> for Error {
    fn from(value: ValidationError) -> Self {
        Self::Validation(value.to_string())
    }
}
```